mountpoint-s3-crt/src/auth/signing_config.rs

```diff
@@ -32,6 +33,49 @@ impl Debug for SigningConfigInner {
     }
 }
 
+impl SigningConfigInner {
+    /// Create a new [SigningConfig] with default options.
+    pub fn new(region: &str, credentials_provider: CredentialsProvider) -> Self {
```

Comment:
Why did we create this constructor for SigningConfigInner when we were doing the same with `default_init_siginig_config()`?

Reply:
Well, some of the configs are now optional. I think this might be easier to read.